cs/test/core/Util.cs

```diff
@@ -271,6 +273,22 @@ public static ArraySegment<byte> SerializeUnsafeCB<T>(T obj)
             return output.Data;
         }
 
+        public static ArraySegment<byte> SerializeUnsafeCBWithPtr<T>(T obj)
```

Comment:
Looking into this and the main issue is now you cannot test compatibility roundtrip between unsafe and safe ptr without introducing all the combinations. Which sounds a bit too complex when the current can deal with it at the expense of having to enter into the delegate to understand what it is happening. What do you think? It is worth the complexity?